src/workloads/sharding/MultiShardTransactions.yml

```diff
@@ -0,0 +1,343 @@
+SchemaVersion: 2018-07-01
+Owner: "@mongodb/sharding"
+Description: |
+
+GlobalDefaults:
+  Database0: &Database0 databaseForMultiShardTestingIndex0
+  Database1: &Database1 databaseForMultiShardTestingIndex1
+
+  # Note that the exact document size may exceed ApproxDocumentSize because of field names and other
+  # fields in the document.
+  ApproxDocumentSize: &ApproxDocumentSize 2000  # = 2kB
+  ApproxDocumentSize50Pct: &ApproxDocumentSize50Pct 1000  # = 1kB
+  DocumentCount: &DocumentCount 100_000
+  NumChunks: &NumChunks 240
+
+  MaxPhase: &MaxPhase 6
+
+Actors:
+- Name: CreateShardedCollection
+  Type: AdminCommand
+  Threads: 1
+  Phases:
+    OnlyActiveInPhases:
+      Active: [0]
+      NopInPhasesUpTo: *MaxPhase
+      PhaseConfig:
+        Repeat: 1
+        Database: admin
+        Operations:
+        - OperationName: AdminCommand
+          OperationCommand:
+            enableSharding: *Database0
+        - OperationName: AdminCommand
+          OperationCommand:
+            enableSharding: *Database1
+        - OperationName: AdminCommand
+          OperationCommand:
+            shardCollection: databaseForMultiShardTestingIndex0.Collection0
+            key: {_id: hashed}
+            numInitialChunks: *NumChunks
+        - OperationName: AdminCommand
+          OperationCommand:
+            shardCollection: databaseForMultiShardTestingIndex1.Collection0
+            key: {_id: hashed}
+            numInitialChunks: *NumChunks
+        - OperationName: AdminCommand
+          OperationCommand:
+            movePrimary: *Database0
+            to: rs0
+        - OperationName: AdminCommand
+          OperationCommand:
+            movePrimary: *Database1
+            to: rs1
+
+- Name: LoadInitialData
+  Type: MonotonicLoader
+  Threads: 4
+  Phases:
+  - &Nop {Nop: true}
+  - Repeat: 1
+    Threads: 4
+    BatchSize: 1000
+    DocumentCount: *DocumentCount
+    CollectionCount: 2
+    Database: *Database0
+    Document: &Document
+      counter: 0
+      padding: {^FastRandomString: {length: {^RandomInt: {min: *ApproxDocumentSize50Pct, max: *ApproxDocumentSize}}}}
+  - Repeat: 1
+    Threads: 4
+    BatchSize: 1000
+    DocumentCount: *DocumentCount
+    CollectionCount: 2
+    Database: *Database1
+    Document: *Document
+  - *Nop
+  - *Nop
+  - *Nop
+  - *Nop
+
+- Name: Transactions
+  Type: CrudActor
+  Threads: 16
+  Database: *Database0
+  Phases:
+    - *Nop
+    - *Nop
+    - *Nop
+    - MetricsName: ShardedCollectionUpdate
+      Duration: 1 minute
+      Collection: Collection0
+      ThrowOnFailure: false  # Transactions are allowed to fail.
+      RecordFailure: true  # We still want to record the metrics even if the transaction is rolled back.
+      Operations:
+      - OperationName: withTransaction
+        OperationCommand:
+          OperationsInTransaction:
```

Comment:
[question] I noticed that we runs 11 update operations in each transaction in this workload. I have some questions:
1. Why don't we make each transaction just run the minimum number of operations it needs to do. For example, for a transaction to "targets two non-sharded collections on different primary shards", it just needs to do an update against `databaseForMultiShardTestingIndex0.Collection1` and an update against `databaseForMultiShardTestingIndex1.Collection1`. 
2. If we want to test coverage for large transaction, we could instead make each update touch multiple documents (e.g. similar to what this workload [does](https://github.com/mongodb/genny/blob/b84fde404e63763634497cacca5f662ed4db6a25/src/workloads/scale/UpdateMillionDocsInTxn.yml#L76)).

Reply:
11 was a mistake. I was going for only 10.

I did it to make sure that most of the operations actually affect multiple shards on sharded collections.

The alternative would be to have not-hashed shard key and query documents from first half and last half of the _id range.
But I don't know how to ensure correct data distribution in that case.

Reply:
> The alternative would be to have not-hashed shard key and query documents from first half and last half of the _id range.
But I don't know how to ensure correct data distribution in that case.

One way is to specify some other field (not _id) as the shard key field, and do manually data distribution by specifying the range for that field (similar to [this](https://github.com/mongodb/genny/blob/48b3fac2e0c8c29b5f196e1263acc8d89e3762ed/src/workloads/sharding/ReshardCollection.yml#L95-L96)) and splitting and moving the chunks accordingly (similar to [this](https://github.com/mongodb/genny/blob/a061a91d04e07bd6da68336e6342ca8187809a6a/src/workloads/sharding/WouldChangeOwningShardBatchWrite.yml#L40-L51)).